test/phoenix_live_view/html_tokenizer_test.exs

```diff
@@ -64,11 +65,75 @@ defmodule Phoenix.LiveView.HTMLTokenizerTest do
 
       assert [
                {:tag_open, "p", [], %{line: 1, column: 1}},
-               {:text, "\n<!--\n<div>\n-->\n", %{line_end: 5, column_end: 1}},
+               {:text, "\n<!--\n<div>\n-->\n",
+                %{line_end: 5, column_end: 1, context: [:comment_start, :comment_end]}},
                {:tag_close, "p", %{line: 5, column: 1}},
                {:tag_open, "br", [], %{line: 5, column: 5}}
              ] = tokenize(code)
     end
+
+    test "adds comment_start and comment_end" do
+      first_part = """
+      <p>
+      <!--
+      <div>
+      """
+
+      {tokens, cont} = HTMLTokenizer.tokenize(first_part, "nofile", 0, [], [], :text)
+
+      assert Enum.reverse(tokens) ==
+               [
+                 {:tag_open, "p", [], %{column: 1, line: 1}},
+                 {:text, "\n<!--\n<div>\n",
+                  %{column_end: 1, line_end: 4, context: [:comment_start]}}
+               ]
+
+      second_part = """
+      </div>
+      -->
+      </p>
+      """
+
+      {tokens, :text} = HTMLTokenizer.tokenize(second_part, "nofile", 0, [], tokens, cont)
+
+      assert Enum.reverse(tokens) == [
+               {:tag_open, "p", [], %{column: 1, line: 1}},
+               {:text, "\n<!--\n<div>\n",
+                %{column_end: 1, line_end: 4, context: [:comment_start]}},
+               {:text, "</div>\n-->\n", %{column_end: 1, context: [:comment_end], line_end: 3}},
```

Comment:
And that is how it would like for cases that the comment ends in the next token.